### DISCUSSION TITLE
SQL ETL soft delete

### DISCUSSION THREAD

**Discussion by djordjedjukic** [ravendb/ravendb#12223] [Q&A] [answered]
Is it possible when a document in RavenDB is deleted, to set the IsDeleted property to true, instead of deleting it?

**Comment by arekpalinski** [comment, marked as answer]
Let me make sure that I understand. You'd like to send a document to SQL database with `IsDeleted: true` instead of propagating the deletion to SQL table?

That's not possible because at the time the ETL process processes the tombstone of a deleted document then that document does not exist.

**Reply by djordjedjukic** [reply]
Yes, that is the case.

Thank you.

**Reply by ayende** [reply]
You can do that using `instead of trigger` on the database side, no?

**Reply by DejanMilicic** [reply]
So something in the lines of https://stackoverflow.com/a/7763409/158320

**Reply by djordjedjukic** [reply]
I will check that. Because the object is complex (1 document from ravendb is mapping on 5 tables in SQL) maybe I will use hard delete in SQL if that is easier.
